types/controlled-frame.d.ts

```diff
@@ -14,170 +14,499 @@
+  matchAboutBlank?: boolean;
+  runAt?: RunAt;
+};
+
+type ClearDataOptions = {
   since?: number;
-}
+};
 
-interface ClearDAtaTypeSet {
-  appcache?: boolean;
+type ClearDataTypeSet = {
   cache?: boolean;
   cookies?: boolean;
   fileSystems?: boolean;
   indexedDB?: boolean;
   localStorage?: boolean;
   persistentCookies?: boolean;
   sessionCookies?: boolean;
-  webSQL?: boolean;
+};
+
+type ZoomMode = 'per-origin' | 'per-view' | 'disabled';
+
+type ImageDetails = {
+  format?: string;
+  quality?: string;
+};
+
+interface ConsoleMessageEvent extends Event {
+  readonly level: number;
+  readonly message: string;
 }
 
-/* https://wicg.github.io/controlled-frame/#html-element */
-interface ControlledFrame extends HTMLElement {
-  src: string;
-  name: string;
-  allowfullscreen: boolean;
-  allowscaling: boolean;
-  allowtransparency: boolean;
-  autosize: boolean;
-  maxheight: string;
-  maxwidth: string;
-  minheight: string;
-  minwidth: string;
-  partition: string;
+type DialogType = 'alert' | 'confirm' | 'prompt';
 
-  readonly contentWindow: WindowProxy;
-  readonly contextMenus: ContextMenus;
+interface DialogController {
+  okay(response?: string): void;
+  cancel(): void;
+}
 
-  // Navigation methods
-  back(): Promise<void>;
-  canGoBack(): boolean;
-  canGoForward(): boolean;
-  forward(): Promise<void>;
-  go(relativeIndex?: number): Promise<void>;
-  reload(): void;
-  stop(): void;
+interface DialogEvent extends Event {
+  readonly messageType: DialogType;
+  readonly messageText: string;
+  readonly dialog: DialogController;
+}
 
-  // Scripting Methods
-  addContentScripts(contentScriptList: ContentScriptDetails[]): Promise<void>;
-  executeScript(details?: InjectDetails | {}): Promise<any>;
-  insertCSS(details?: InjectDetails): Promise<void>;
-  removeContentScripts(scriptNameList: string[]): Promise<void>;
+type WindowOpenDisposition =
+  | 'ignore'
+  | 'save_to_disk'
+  | 'current_tab'
+  | 'new_background_tab'
+  | 'new_foreground_tab'
+  | 'new_window'
+  | 'new_popup';
 
-  // Configuration methods
-  clearData(options: ClearDataOptions = {}): Promise<void>;
-  getAudioState(): Promise<boolean>;
-  getZoom(): Promise<number>;
-  isAudioMuted(): Promise;
-  setAudioMuted(mute: boolean): void;
-  setZoom(zoomFactor: number): Promise<void>;
+interface NewWindowController {
+  attach(newControlledFrame: HTMLControlledFrameElement): void;
+  discard(): void;
+}
 
-  // Capture methods
-  captureVisibleRegion(): void;
-  print(): void;
+interface NewWindowEvent extends Event {
+  readonly window: NewWindowController;
+  readonly targetUrl: string;
+  readonly name: string;
+  readonly windowOpenDisposition: WindowOpenDisposition;
+  readonly partition: string;
+  readonly initialHeight: number;
+  readonly initialWidth: number;
+}
+
+type PermissionType =
+  | 'media'
+  | 'geolocation'
+  | 'pointerLock'
+  | 'download'
+  | 'filesystem'
+  | 'fullscreen'
+  | 'hid';
+
+interface PermissionRequestController {
```

Comment:
is the idea here to have one common type (as opposed to all the different types in the spec, such as `MediaPermissionRequestController`, `GeolocationPermissionRequestController`, ...)?